contracts/interchain-token/InterchainToken.sol

```diff
@@ -62,7 +61,10 @@ abstract contract InterchainToken is IInterchainToken, ERC20 {
          * @dev if you know the value of `tokenManagerRequiresApproval()` you can just skip the if statement and just do nothing or _approve.
          */
         if (tokenManagerRequiresApproval()) {
-            _approve(sender, address(tokenManager), allowance[sender][address(tokenManager)] + amount);
+            uint256 allowance_ = allowance[sender][address(tokenManager)];
+            if(allowance_ != type(uint256).max) {
+                _approve(sender, address(tokenManager), allowance_ + amount);
+            }
```

Comment:
```suggestion
            if (allowance_ > type(uint256).max - amount) {
                allowance_ = type(uint256).max - amount
            }
            
            _approve(sender, address(tokenManager), allowance_ + amount);
```
maybe something like this so it's more robust against overflows?

Reply:
That's not even checked in the ERC20 implementation as type(uint256).max is considered too large.

Reply:
Hmm, but in our case, we're adding to an existing value vs just overwriting to a provided uint256. Not a strong opinion though